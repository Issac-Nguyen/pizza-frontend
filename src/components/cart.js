import React from 'react';
import {Badge} from 'antd';
import { connect } from 'react-redux';
import Icon from './Icon';
import {showCart} from '../actions/userActions'
import { useState } from 'react';

export const Cart = (props) => {
    return (
        <div className="cart-ico" onClick={props.showCart}>
                <Badge count={props.book.length}>
                    <Icon name="la-shopping-cart" />
                </Badge>
        </div>
    )
}

const mapStateToProps = state => ({
    ...state
   })

const mapDispatchToProps = dispatch => ({
    showCart: () => {dispatch(showCart())}
})

export default connect(mapStateToProps, mapDispatchToProps)(Cart);