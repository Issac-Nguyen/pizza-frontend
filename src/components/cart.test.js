import React from 'react'
import {render, cleanup} from '@testing-library/react'
import {Cart} from './cart'

describe('Cart', () => {
    afterEach(cleanup);

    it('should take a snapshot', () => {
       const { asFragment } = render(<Cart book={[]}/>)
       
       expect(asFragment(<Cart book={[]}/>)).toMatchSnapshot()
    })
    

    it('should have icon', () => {
        const { container  } = render(<Cart book={[]}/>)
        
        expect(container.querySelector('i')).toBeInTheDocument();
        expect(container.querySelector('i')).toHaveClass('las la-shopping-cart');
        // expect(container.querySelector('Icon')).toHaveClass('las la-shopping-cart');
     })

     it('should take snapshot with length', () => {
        const { asFragment } = render(<Cart book={[{}]}/>)
        expect(asFragment(<Cart book={[{}]}/>)).toMatchSnapshot()
     })
})
