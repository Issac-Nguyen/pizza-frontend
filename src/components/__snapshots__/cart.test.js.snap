// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="cart-ico"
  >
    <span
      class="ant-badge"
    >
      <i
        class="las la-shopping-cart"
        data-testid="icon"
      />
    </span>
  </div>
</DocumentFragment>
`;

exports[`Cart should take snapshot with length 1`] = `
<DocumentFragment>
  <div
    class="cart-ico"
  >
    <span
      class="ant-badge"
    >
      <i
        class="las la-shopping-cart"
        data-testid="icon"
      />
      <sup
        class="ant-scroll-number ant-badge-count ant-badge-zoom-appear"
        data-show="true"
        title="1"
      >
        <span
          class="ant-scroll-number-only"
          style="transition: none; transform: translateY(-1100%);"
        >
          <p
            class="ant-scroll-number-only-unit"
          >
            0
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            1
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            2
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            3
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            4
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            5
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            6
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            7
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            8
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            9
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            0
          </p>
          <p
            class="ant-scroll-number-only-unit current"
          >
            1
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            2
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            3
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            4
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            5
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            6
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            7
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            8
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            9
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            0
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            1
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            2
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            3
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            4
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            5
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            6
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            7
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            8
          </p>
          <p
            class="ant-scroll-number-only-unit"
          >
            9
          </p>
        </span>
      </sup>
    </span>
  </div>
</DocumentFragment>
`;
